src/classes/CON_ContactMerge_TDTM.cls

```diff
@@ -39,159 +39,333 @@
+        private Map<Id, Id> oldIndividualAcctIdByMasterId = new Map<Id, Id>();
+        private Map<Id, Id> newIndividualAcctIdByMasterId = new Map<Id, Id>();
+
+        /***************************************************************************************************************
+         * @description Fixes records involved in the merge process
+         * @param mergeKeys - the ids releveant to the merge process
+         */
+        private void fixup(List<String> mergeKeys) {
+            setMergeKeyIds(mergeKeys);
+
+            List<Account> oldAccounts = getAccounts(oldAccountIds);
+            List<Contact> masterContacts = getContacts(masterContactIds);
+
+            householdIds = buildHouseholdIds(oldAccounts);
+            householdIds.addAll(buildHouseholdIds(masterContacts));
+
+            oldIndividualAcctIdByMasterId = buildAcctIdByMasterId(oldAccounts);
+            newIndividualAcctIdByMasterId = buildAcctIdByMasterId(masterContacts);
+
+            fixupAccountData();
+            fixupContactData(masterContacts);
         }
-    
-        // get the winning Contacts (with the fields we need)
-        RLLP_OppRollup_UTIL oppRollupUtil = new RLLP_OppRollup_UTIL(); 
-        string strSoql = oppRollupUtil.buildContactQuery();
-        strSoql = strSoql.replace('from Contact', '');
-        strSoql += ', AccountId, Account.npe01__SYSTEM_AccountType__c, Primary_Affiliation__c from Contact where id IN :listContactIdWinner';
-        list<Contact> listContactWinner = Database.query(strSoql);
-
-        // create maps of contactId's to AccountId's that some of our routines need.
-        map<Id,Id> mapContactIdAccountIdWinner = new map<Id,Id>();
-        map<Id,Id> mapContactIdAccountIdLoser = new map<Id,Id>();
-        list<Id> listAccIdHH= new list<Id>();
-        for (Contact con : listContactWinner) {
-        	if (con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.HH_ACCOUNT_TYPE)
-        	   listAccIdHH.add(con.AccountId);
-            if (con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.HH_ACCOUNT_TYPE ||
-                con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE ||
-                con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.BUCKET_ORGANIZATION_TYPE) {
-                mapContactIdAccountIdWinner.put(con.Id, con.AccountId);        
+
+        /*******************************************************************************************************
+        * @description Returns the account Id of accounts with household account types
+        * @param accounts - The accounts to return the Id for when they are a household account
+        * @return List<Id> - The list of household account Ids
+        ********************************************************************************************************/
+        @TestVisible
+        private List<Id> buildHouseholdIds(List<Account> accounts) {
```

Comment:
Suggestion: since `build` usually indicates building an sObject but not inserting it into the DB, maybe we can use `extract` or any other keyword indicating Ids of specific input Accounts will be returned?